actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -93,6 +93,33 @@ def assert_enqueued_emails(number, &block)
       assert_enqueued_jobs number, only: [ ActionMailer::DeliveryJob, ActionMailer::Parameterized::DeliveryJob ], &block
     end
 
+    # Asserts that a specific email has been enqueued, optionally
+    # also matching arguments.
+    #
+    #   def test_emails
+    #     ContactMailer.welcome.deliver_later
+    #     assert_enqueued_email_with ContactMailer, :welcome
+    #   end
+    #
+    # If a block is passed, that block should cause the specified email
+    # to be enqueued.
+    #
+    #   def test_emails_again
+    #     assert_enqueued_email_with ContactMailer, :welcome
+    #       ContactMailer.welcome.deliver_later
+    #     end
+    #   end
+    def assert_enqueued_email_with(mailer, method, args: nil, queue: "mailers", &block)
+      args = args && [mailer.to_s, method.to_s, "deliver_now", *args]
+
+      assert_enqueued_with(
+        job: ActionMailer::DeliveryJob,
```

Comment:
Mailers can also be enqueued with `ActionMailer::Parameterized::DeliveryJob`.

Reply:
Thanks, @kaspth! Suggestions on how to handle both are very welcome.

Reply:
You could add a separate helper: `assert_enqueued_parameterized_email_with`.

Alternatively, this helper could assert that an `ActionMailer::Parameterized::DeliveryJob` is enqueued if `args` is Hash-like or responds to `to_hash`.